server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -199,14 +302,14 @@ boolean isHeapUsageDominatedBySearch(List<SearchShardTask> searchShardTasks) {
     }
 
     /**
-     * Filters and returns the list of currently running SearchShardTasks.
+     * Filters and returns the list of currently running tasks of specified type.
      */
-    List<SearchShardTask> getSearchShardTasks() {
+    <T extends CancellableTask & SearchBackpressureTask> List<CancellableTask> getTaskByType(Class<T> type) {
```

Comment:
> Can you just update `SearchBackpressureTask` to extend `CancellableTask` instead?

`CancellableTask` is an abstract class, whereas `SearchBackpressureTask` is an interface. It cannot extend the class.
Also, the return type cannot be `List<SearchBackpressureTask>` because we use the result of this method multiple places, i.e. `TaskResourceUsageTracker.stat()`, which accepts `List<? extends Task>` as an argument. However, `SearchBackpressureTask` does not extend `Task`.